addons/product_margin/tests/test_product_margin.py

```diff
@@ -0,0 +1,118 @@
+
+        self.AccountInvoice.create({
+            'partner_id': self.ref("base.res_partner_1"),
+            'account_id': account_payable,
+            'type': 'in_invoice',
+            'company_id': self.ref('base.main_company'),
+            'name': 'invoice to supplier',
+            'invoice_line_ids': [(0, 0, {
+                        'name': self.Product.name,
+                        'account_id': account_expense,
+                        'product_id': self.Product.id,
+                        'quantity': 10.0,
+                        'uom_id': self.ref('product.product_uom_unit'),
+                        'price_unit': 300.00,
+                    })],
+        }).action_invoice_open()
+
+        self.AccountInvoice.create({
+            'partner_id': self.ref("base.res_partner_2"),
+            'account_id': account_payable,
+            'type': 'in_invoice',
+            'company_id': self.ref('base.main_company'),
+            'name': 'invoice to supplier',
+            'invoice_line_ids': [(0, 0, {
+                        'name': self.Product.name,
+                        'product_id': self.Product.id,
+                        'quantity': 4.0,
+                        'account_id': account_expense,
+                        'uom_id': self.ref('product.product_uom_unit'),
+                        'price_unit': 450.00
+                    })],
+        }).action_invoice_open()
+
+        # Define Customer Invoice
+        self.AccountInvoice.create({
+            'partner_id': self.ref("base.res_partner_1"),
+            'name': 'invoice to client',
+            'type': 'out_invoice',
+            'company_id': self.ref('base.main_company'),
+            'account_id': account_receivable,
+            'invoice_line_ids': [(0, 0, {
+                        'name': self.Product.name,
+                        'product_id': self.Product.id,
+                        'quantity': 20.0,
+                        'account_id': account_revenue,
+                        'uom_id': self.ref('product.product_uom_unit'),
+                        'price_unit': self.Product.list_price
+                    })],
+        }).action_invoice_open()
+
+        self.AccountInvoice.create({
+            'partner_id': self.ref("base.res_partner_2"),
+            'name': 'invoice to client',
+            'type': 'out_invoice',
+            'company_id': self.ref('base.main_company'),
+            'account_id': account_receivable,
+            'invoice_line_ids': [(0, 0, {
+                        'name': self.Product.name,
+                        'product_id': self.Product.id,
+                        'quantity': 10.0,
+                        'account_id': account_revenue,
+                        'uom_id': self.ref('product.product_uom_unit'),
+                        'price_unit': 550.00
+                    })],
+        }).action_invoice_open()
+
+        result = self.Product._compute_product_margin_fields_values(self)
```

Comment:
That seems a bit nonsensical, we're passing a TestCase instance as `field_names` parameter to `_compute_product_margin_fields_values`?